File: 1233. Remove Sub-Folders from the Filesystem/Solution.cpp
#include <algorithm>
#include <ranges>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder)
    {
        ranges::sort(folder);
        vector<string> ans{folder[0]};
        for (size_t i = 1; i < folder.size(); ++i) {
            string& s = folder[i];
            string& last = ans.back();
            if (!s.starts_with(last) || s[last.size()] != '/')
                ans.emplace_back(s);
        }
        return ans;
    }
};

File: 1233. Remove Sub-Folders from the Filesystem/Solution.py
class Solution:
    def removeSubfolders(self, folder: list[str]) -> list[str]:
        folder.sort()
        ans = [folder[0]]
        for s in folder[1:]:
            last = ans[-1]
            if not s.startswith(last) or s[len(last)] != "/":
                ans.append(s)
        return ans
